src/mlpack/bindings/go/CMakeLists.txt

```diff
@@ -0,0 +1,247 @@
+macro (go_not_found_return message)
+  message(STATUS "${message}")
+  macro (add_go_binding name)
+    # Do nothing.
+  endmacro ()
+
+  return()
+endmacro ()
+
+# If we are not supposed to make Python bindings, define the macro so it does
+# nothing and leave this file.
+if (NOT BUILD_GO_BINDINGS)
+  go_not_found_return("Not building Go bindings.")
+endif ()
+
+find_package(Go)
+if (NOT GO_FOUND)
+  go_not_found_return("Go not found; not building Go bindings.")
+else ()
+  message(STATUS "Found Go: ${GO_EXECUTABLE}")
+endif ()
```

Comment:
Can you please help with this I am not able to figure out the problem here?

Reply:
You could reproduce this by configuring like this, but without Go installed on the system:

```
cmake -DBUILD_GO_BINDINGS=ON ../
```

What you will find is that no error is issued, and the Go bindings aren't built, but there is no obvious error message indicating so.  So we should find a way so that if the user specifies that Go bindings should be built, then it should be an error if Go is not found.  Let me know if I can clarify that.

Reply:
When I remove Golang from my system and tried using `cmake -DBUILD_GO_BINDINGS=ON ../ `
I got this error for golang:
```
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp  
-- Found Git: /usr/bin/git (found version "2.7.4") 
-- gitversion.hpp is already up to date.
-- Not building Markdown bindings.
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Found Python: /usr/bin/python
-- Found PY_distutils: /usr/lib/python2.7/distutils  
-- Found PY_Cython: /home/asus/.local/lib/python2.7/site-packages/Cython  
-- Found PY_numpy: /usr/lib/python2.7/dist-packages/numpy  
-- Found PY_pandas: /usr/lib/python2.7/dist-packages/pandas  
-- Could NOT find Go (missing:  GO_EXECUTABLE) 
-- Go not found; not building Go bindings.
```

Reply:
Right, but did the configuration fail overall?  Or did it continue and successfully configure the project?  My point is, if the user specifies `BUILD_GO_BINDINGS=ON` but Go is not found, it should be a fatal error; configuration should not complete successfully.  This is because I have found users are unlikely to go back and review the CMake output to figure out what is going on. :)

Reply:
No it didn't I will look into it and let you know.
Thank You

Reply:
I have made changes as suggested can you please take a look.
Thank You.

Reply:
Looks good---I tested that if `BUILD_GO_BINDINGS=ON` and Go is not installed, that the build does fail. :+1: